lib/easybill-rest-client/api_client.rb

```diff
@@ -26,55 +29,58 @@ def request_collection(method, endpoint, params = {})
     end
 
     def request(method, endpoint, params = {})
-      Retryable.retryable(
-        tries: tries,
-        sleep: retry_cool_off_time,
-        on: EasybillRestClient::TooManyRequests
-      ) do
-        response = perform_request(method, endpoint, params)
-        process_response(response)
+      @request_logger = RequestLogger.new(logger: logger, request_id: Time.now.to_f)
```

Comment:
Love the idea with the RequestId 👍 

How about using `SecureRandom.hex` or similar instead? IMHO it's way easier to tell 2 requests apart:

```
1468334559.878815
1468334560.7273018

ade0c8fe49aab8738f4bff37e7e31ec7
125fd30c944401a0c318c805e559965b
```